tests/bitset.test.cpp

```diff
@@ -0,0 +1,202 @@
+#include <boost/ut.hpp>
+#include <cstdint>
+#include <libxbitset/bitset.hpp>
+
+namespace embed {
+boost::ut::suite bitset_test = [](){
+    using namespace boost::ut;
+
+    //TEST SET() FOR INITIAL VALUE OF 1
+    "bitset::set()_set_no_change_1"_test = [](){
+        xstd::bitset<int> test_set(1); 
+        //maintain intial value for odd number using set(0,true)
+        expect(that % 0x0000'0001 == test_set.set(0,true)); 
+        expect(that % 0x0000'0001 == test_set.set(0,true));
+    };
+
+    "bitset::set()_increment_position_1"_test = [](){
+        xstd::bitset<int> test_set(1);
+        //incrementing position with same value
+        expect(that % 0x0000'0001 == test_set.set(0,true)); 
+        expect(that % 0x0000'0003 == test_set.set(1,true));
+        expect(that % 0x0000'0007 == test_set.set(2,true));
+        expect(that % 0x0000'000F == test_set.set(3,true));
+    };
+
+    "bitset::set()_increment_value_1"_test = [](){
+        xstd::bitset<int> test_set(1);
+        //incrementing value with same position
+        expect(that % 0x0000'0000 == test_set.set(0,false));
+        expect(that % 0x0000'0001 == test_set.set(0,true));
+    };
+
+    //TEST SET() FOR INITIAL VALUE OF 2
+    "bitset::set()_set_no_change_2"_test = [](){
+        xstd::bitset<int> test_set(2); 
+        //maintain initial value for even number using set(0,false)
+        expect(that % 0x0000'0002 == test_set.set(0,false));
+        expect(that % 0x0000'0002 == test_set.set(0,false));
+    };  
+
+    "bitset::set()_increment_position_2"_test = [](){
+        xstd::bitset<int> test_set(2);
+        //incrementing position with same value
+        expect(that % 0x0000'0003 == test_set.set(0,true));
+        expect(that % 0x0000'0003 == test_set.set(1,true));
+        expect(that % 0x0000'0007 == test_set.set(2,true));
+        expect(that % 0x0000'000F == test_set.set(3,true));
```

Comment:
```suggestion
        expect(that % 0x0000'0003 == test_set.set(0,true));
        expect(that % 0x0000'0003 == test_set.set(1,true));
        expect(that % 0x0000'0007 == test_set.set(2,true));
        expect(that % 0x0000'000F == test_set.set(3,true));
```

In cases where it defaults to true, which it does, you don't need to put true every time. We want our unit tests to also be documentation of how a user would use the code. So lets eliminate the unnecessary stuff.